packages/muelu/src/Graph/Containers/MueLu_Zoltan2GraphAdapter.hpp

```diff
@@ -0,0 +1,459 @@
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact
+//                    Jonathan Hu       (jhu@sandia.gov)
+//                    Andrey Prokopenko (aprokop@sandia.gov)
+//                    Ray Tuminaro      (rstumin@sandia.gov)
+//
+// ***********************************************************************
+//
+// @HEADER
+
+#ifndef MUELU_ZOLTAN2GRAPHADAPTER_HPP_
+#define MUELU_ZOLTAN2GRAPHADAPTER_HPP_
+
+#include "MueLu_ConfigDefs.hpp"
+
+#if defined(HAVE_MUELU_ZOLTAN2) && defined(HAVE_MPI)
+
+#include <Teuchos_RCP.hpp>
+#include <Teuchos_Comm.hpp>
+#include <Teuchos_ArrayView.hpp>
+#include <Xpetra_Map.hpp>
+#include <Zoltan2_InputTraits.hpp>
+#include <Zoltan2_GraphAdapter.hpp>
+#include <Zoltan2_StridedData.hpp>
+#include <Zoltan2_PartitioningSolution.hpp>
+#include "MueLu_GraphBase.hpp"
+
+
+
+// Zoltab2 InputTraits for MueLu Graph objects
+namespace Zoltan2 {
+
+template <typename LocalOrdinal,
+          typename GlobalOrdinal,
+          typename Node>
+struct InputTraits<MueLu::GraphBase<LocalOrdinal,GlobalOrdinal,Node> >
+{
+  typedef Zoltan2::default_scalar_t scalar_t;
+  typedef LocalOrdinal  lno_t;
+  typedef GlobalOrdinal gno_t;
+  typedef size_t offset_t;
+  typedef Zoltan2::default_part_t  part_t;
+  typedef Node          node_t;
+  static inline std::string name() {return "MueLu::Graph";}
+
+  Z2_STATIC_ASSERT_TYPES // validate the types
+};
+}//end namespace Zoltan2
+
+
+namespace MueLu {
+
+template <typename User, typename UserCoord=User>
+class MueLuGraphBaseAdapter : public Zoltan2::GraphAdapter<User,UserCoord> {
```

Comment:
Without checking any details: how is this different from the Zoltan2 Xpetra adapter?

Reply:
That requires an Xpetra:CrsGrap under the hood, which GrapBase is *not*